# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../runit/rockify.patch
# Copyright (C) 2004 - 2008 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

we need to export vars to be read inside init.d/system
currently we simulate sysv's stage 2
  Alejandro

--- ./etc/debian/1.orig	2005-08-24 16:14:38.000000000 -0400
+++ ./etc/debian/1	2005-09-24 18:42:18.000000000 -0400
@@ -1,10 +1,17 @@
 #!/bin/sh
 # system one time tasks
 
-PATH=/command:/sbin:/bin:/usr/sbin:/usr/bin
+{
+export PATH=/command:/sbin:/bin:/usr/sbin:/usr/bin
 
-/etc/init.d/rcS
-/etc/init.d/rmnologin
+for x in /etc/runit/1.d/*; do
+	[ -x $x ] && $x start
+done
 
-touch /etc/runit/stopit
-chmod 0 /etc/runit/stopit
+# write ET mark for btee
+echo -ne '\004'
+
+} 2>&1 | /sbin/btee a /var/log/init.msg
+
+touch /etc/runit/{stopit,reboot}
+chmod 0 /etc/runit/{stopit,reboot}
--- ./etc/debian/3.orig	2006-06-29 07:53:16.000000000 +0000
+++ ./etc/debian/3	2006-07-07 13:09:50.000000000 +0000
@@ -1,14 +1,18 @@
 #!/bin/sh
 exec 2>&1
 
-PATH=/command:/sbin:/bin:/usr/sbin:/usr/bin
-
-LAST=0
-test -x /etc/runit/reboot && LAST=6
+export PATH=/command:/sbin:/bin:/usr/sbin:/usr/bin
+  
+export INIT_DOES_REBOOTHALT=1
+export RUNLEVEL=0
+test -x /etc/runit/reboot && export RUNLEVEL=6
 
 echo 'Waiting for services to stop...'
-sv -w196 force-stop /service/*
-sv exit /service/*
+sv -w196 force-stop /var/service/*
+sv exit /var/service/*
 
 echo 'Shutdown...'
-/etc/init.d/rc $LAST
+for x in /etc/runit/3.d/*; do
+      [ -x $x ] && $x stop
+done
+true
--- ./etc/debian/ctrlaltdel.orig	2005-08-24 16:14:38.000000000 -0400
+++ ./etc/debian/ctrlaltdel	2005-09-14 05:05:04.000000000 -0400
@@ -4,6 +4,6 @@
 MSG="System is going down in 14 seconds..."
 
 # echo 'disabled.' ; exit
-touch /etc/runit/stopit
-chmod 100 /etc/runit/stopit && echo "$MSG" | wall
+touch /etc/runit/{stopit,reboot}
+chmod 100 /etc/runit/{stopit,reboot} && echo "$MSG" | wall
 /bin/sleep 14
--- ./etc/debian/2.orig	2006-03-12 22:46:12.000000000 +0100
+++ ./etc/debian/2	2006-03-12 22:46:26.000000000 +0100
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-PATH=/command:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
+PATH=/command:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11/bin
 
 exec env - PATH=$PATH \
-runsvdir -P /service 'log: ...........................................................................................................................................................................................................................................................................................................................................................................................................'
+runsvdir -P /var/service 'log: ...........................................................................................................................................................................................................................................................................................................................................................................................................'
